### ISSUE TITLE
Failing CNCF test due to Cilium

### ISSUE THREAD

**Issue by T-Kukawka** [giantswarm/roadmap#3272]

**Comment by T-Kukawka** [comment]
cc @puja108 @luca-rui @JosephSalisbury

**Comment by JosephSalisbury** [comment]
See https://github.com/cilium/cilium/pull/29524 for where cilium drop the test from their conformance testing

**Comment by JosephSalisbury** [comment]
and see https://github.com/cncf/k8s-conformance/pull/3049#issuecomment-1956107626 - eks anywhere is hitting the same problem we are (maybe we can be friends)

**Comment by QuentinBisson** [comment]
The test was promoted in 1.29 and we are running them for 1,25 so I guess we should be able to exclude the test and still be conformant right?

**Comment by yulianedyalkova** [comment]
Can @giantswarm/team-cabbage look into this one and see if there is any solution by now?

**Comment by weatherhog** [comment]
According to this https://github.com/cilium/cilium/pull/33434 the issue got fixed and also back ported to cilium 1.16

**Comment by yulianedyalkova** [comment]
Has that fix reached our product because the tests are still [failing](https://gigantic.slack.com/archives/C0561JVB0HY/p1736745771810199) on our end?

**Comment by mcharriere** [comment]
> Has that fix reached our product because the tests are still [failing](https://gigantic.slack.com/archives/C0561JVB0HY/p1736745771810199) on our end?

No, we were waiting for v30 to include Cilium v1.16. As now all minor releases are using cilium 1.15.6 we can update.
I created [this PR](https://github.com/giantswarm/releases/pull/1536) relaxing this constrain so next minor releases should include it. Are you planning on releasing it any time soon?

**Comment by yulianedyalkova** [comment]
> Are you planning on releasing any time soon?

v30 is on the way as soon as we can get back to it so if we get the tests passing there, that'd be perfect. Thank you!

**Comment by mcharriere** [comment]
Sorry it seems there was a confusion around this. The fix is NOT present on any v1.16.x. This means we need to keep on waiting until [v1.17.x](https://github.com/cilium/cilium/releases/tag/v1.17.0)

> 🔀 Load Balance based on L4 Protocol: Differentiate TCP and UDP based protocols for load balancing, so multiple services on the same port can be directed to different backends (https://github.com/cilium/cilium/pull/33434, @jibi)

Unfortunately, unless we move cilium v1.16 to v29, we can't introduced v1.17 in v30.

**Comment by yulianedyalkova** [comment]
Oh, this is far from ideal. We have v30 ready to ship and we have to do it _asap_ as Kubernetes v1.29 is EOF in 3 days.
